File: frontend/src/types/configuration.ts
export interface ConfigurationDefinition {
  id: number
  key: string
  description?: string
  type?: string // INTEGER, ENUM, BOOLEAN, STRING
  allowedValues?: string[]
  defaultValue?: string
}

export interface Configuration {
  id: number
  organisation?: { orgId: number; name?: string }
  definition: ConfigurationDefinition
  value?: string | null
  createdAt?: string
  updatedAt?: string
}

export interface ConfigurationCreateRequest {
  definition: { key: string }
  value?: string | null
}

export interface ConfigurationUpdateRequest {
  value?: string | null
}

File: src/main/java/com/taceiq/entity/Configuration.java
package com.taceiq.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;

@Entity
@Table(name = "configurations", uniqueConstraints = @UniqueConstraint(columnNames = {"org_id", "definition_id"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Configuration {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "org_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "users", "roles", "configurations"})
    private Organisation organisation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "definition_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private ConfigurationDefinition definition;

    // Organisation-selected value (must be in definition.allowedValues)
    @Column(name = "config_value", columnDefinition = "TEXT")
    private String value;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    void onUpdate() {
        this.updatedAt = Instant.now();
    }
}
